chpl/chpl-service/src/main/java/gov/healthit/chpl/scheduler/job/ModifyTestFunctionalityJob.java

```diff
@@ -0,0 +1,103 @@
+import org.quartz.JobExecutionException;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.security.core.context.SecurityContextHolder;
+import org.springframework.stereotype.Component;
+import org.springframework.transaction.annotation.Transactional;
+import org.springframework.web.context.support.SpringBeanAutowiringSupport;
+
+import gov.healthit.chpl.auth.permission.GrantedPermission;
+import gov.healthit.chpl.auth.user.JWTAuthenticatedUser;
+import gov.healthit.chpl.dao.impl.BaseDAOImpl;
+import gov.healthit.chpl.entity.TestFunctionalityEntity;
+import gov.healthit.chpl.exception.EntityRetrievalException;
+import gov.healthit.chpl.manager.TestingFunctionalityManager;
+
+public class ModifyTestFunctionalityJob extends QuartzJob {
+    private static final Logger LOGGER = LogManager.getLogger("modifyTestFunctionalityJobLogger");
+    private static final String TF_OLD_NUMBER = "(b)(3)(iii)";
+    private static final String TF_NEW_NUMBER = "(b)(3)(C)";
+    private static final String TF_NEW_NAME = "Optional: 170.315(b)(3)(C) For each transaction listed in paragraph (b)(3)(A) of this section, the technology must be able to receive and transmit the reason for the prescription using the indication elements in the SIG Segment";
+
+    @Autowired
+    private ModifyableTestFunctionalityDao modifyableTestFunctionalityDao;
+
+    @Autowired
+    private TestingFunctionalityManager testingFunctionalityManager;
+
+    @Override
+    public void execute(JobExecutionContext jobContext) throws JobExecutionException {
+        LOGGER.info("********* Starting the Test Functionality Modification job. *********");
+
+        SpringBeanAutowiringSupport.processInjectionBasedOnCurrentContext(this);
+        setSecurityContext();
+        try {
+            modifyTestFunctionality();
+            testingFunctionalityManager.onApplicationEvent(null);
+            LOGGER.info("Updated Test Functionality");
+        } catch (EntityRetrievalException e) {
+            LOGGER.error("Unable to update Test Functionality: " + e.getMessage());
+        }
+
+        LOGGER.info("********* Completed the Test Functionality Modification job. *********");
+    }
+
+    private void modifyTestFunctionality() throws EntityRetrievalException {
+        modifyableTestFunctionalityDao.update(TF_OLD_NUMBER, TF_NEW_NUMBER, TF_NEW_NAME);
+    }
+
+    private void setSecurityContext() {
+        JWTAuthenticatedUser adminUser = new JWTAuthenticatedUser();
+        adminUser.setFullName("Administrator");
+        adminUser.setId(-2L);
+        adminUser.setFriendlyName("Admin");
+        adminUser.setSubjectName("admin");
+        adminUser.getPermissions().add(new GrantedPermission("ROLE_ADMIN"));
+
+        SecurityContextHolder.getContext().setAuthentication(adminUser);
+        SecurityContextHolder.setStrategyName(SecurityContextHolder.MODE_INHERITABLETHREADLOCAL);
+    }
+
+    @Component("modifyableTestFunctionalityDao")
```

Comment:
Not important in the least for the PR, but I think the spelling is modifiable

Reply:
Thanks! It definitely looked odd